### PR TITLE
[indexer] - suix_getEpochs and suix_getCurrentEpoch APIs

### PR DESCRIPTION
**Pull request by patrickkuo**
## Description 

* store system_state
* store epoch
* added get_epochs and get_current_epoch methods.

## Test Plan 
added integration test for getting epochs

### CODE REVIEW

File: crates/sui-indexer/src/apis/extended_api.rs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+}
+
+#[async_trait]
+impl<S: IndexerStore + Sync + Send + 'static> ExtendedApiServer for ExtendedApi<S> {
+    async fn get_epoch(
+        &self,
+        cursor: Option<EpochId>,
+        limit: Option<usize>,
+    ) -> RpcResult<EpochPage> {
+        let limit = cap_page_limit(limit);
```
Comment (healthydeve) [inline crates/sui-indexer/src/apis/extended_api.rs:33]: nit: i think we are moving to `validate_limit` method to keep it consistent

File: crates/sui-indexer/migrations/2023-03-20-041133_epoch/down.sql [hunk 2]
```diff
@@ -0,0 +1 @@
+-- This file should undo anything in `up.sql`
```
Comment (gegaowp) [inline crates/sui-indexer/migrations/2023-03-20-041133_epoch/down.sql:1]: DROP TABLE?

File: crates/sui-indexer/migrations/2023-03-20-042226_system_state/down.sql [hunk 3]
```diff
@@ -0,0 +1 @@
+-- This file should undo anything in `up.sql`
```
Comment (gegaowp) [inline crates/sui-indexer/migrations/2023-03-20-042226_system_state/down.sql:1]: DROP ?

File: crates/sui-indexer/migrations/2023-03-20-041133_epoch/up.sql [hunk 4]
```diff
@@ -0,0 +1,24 @@
+-- Your SQL goes here
```
Comment (gegaowp) [inline crates/sui-indexer/migrations/2023-03-20-041133_epoch/up.sql:1]: nit: remove?

File: crates/sui-indexer/src/store/pg_indexer_store.rs [hunk 5]
```diff
@@ -1021,13 +1037,149 @@ impl IndexerStore for PgIndexerStore {
...
+            .serializable()
+            .read_write()
+            .run(|conn| {
+                if let Some(last_epoch) = &data.last_epoch {
+                    diesel::insert_into(epochs::table)
+                        .values(last_epoch)
+                        .on_conflict(epochs::epoch)
+                        .do_update()
+                        .set((
+                            epochs::last_checkpoint_id.eq(excluded(epochs::last_checkpoint_id)),
```
Comment (gegaowp) [inline crates/sui-indexer/src/store/pg_indexer_store.rs:1054]: not in this PR, but same thing as object updates, would be great if we have ways to test/even better ensure these cover all columns ..

File: crates/sui-indexer/src/store/pg_indexer_store.rs [hunk 6]
```diff
@@ -1007,6 +1011,18 @@ impl IndexerStore for PgIndexerStore {
             .serializable()
             .read_write()
             .run(|conn| {
+                // update epoch transaction count
+                let sql = "UPDATE epochs e1
```
Comment (gegaowp) [inline crates/sui-indexer/src/store/pg_indexer_store.rs:1015]: can we also update some other accumulative metrics like gas etc. as long as checkpoints have it?
```
    pub total_gas_cost: i64,
    pub total_computation_cost: i64,
    pub total_storage_cost: i64,
    pub total_storage_rebate: i64,
```

**Comment by gegaowp** [comment]
overall looks greate!

File: crates/sui-indexer/migrations/2023-03-20-041133_epoch/down.sql (hunk 2 above)
Comment (patrickkuo) [reply crates/sui-indexer/migrations/2023-03-20-041133_epoch/down.sql:1]: oops forgot

File: crates/sui-indexer/src/store/pg_indexer_store.rs (hunk 6 above)
Comment (patrickkuo) [reply crates/sui-indexer/src/store/pg_indexer_store.rs:1015]: the epoch tx count was a requirement from the FE team..... lets add them if there are requirement, the epoch info response is quite big already....

Comment (gegaowp) [reply crates/sui-indexer/src/store/pg_indexer_store.rs:1015]: got it, that's fair, even though I feel updating one column is roughly the same as updating several regarding machine cost LOL

**Review by gegaowp — APPROVED** [review]

**Pull request merged**